File: cpp/0700二叉搜索树中的搜索.cpp
#include"mybitree.h"

TreeNode *searchBST(TreeNode *root, int val)
{
    while(root)
    {
        if(root->val==val)
            return root;
        if(val<root->val)
            root = root->left;
        else
            root = root->right;
    }

    return NULL;
}

File: python/main/0700二叉搜索树中的搜索.py
from binarytree import TreeNode

def searchBST(root: TreeNode, val: int) -> TreeNode:
    if(not root):
        return None
    if(root.val==val):
        return root

    if(root.val>val):
        return searchBST(root.left,val)
    else:
        return searchBST(root.right,val)
